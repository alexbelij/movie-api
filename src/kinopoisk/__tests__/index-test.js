/* @flow */

import {
  modelFromObject,
  modelFromFirstElement,
  modelFromFirstElementAtPath,
} from '../../test-utils';
import Kinopoisk from '../';

const filmId = 714888; // Star Wars: The Force Awakens
const movieQuery = {
  title: 'Звёздные войны: Пробуждение силы',
};
const tvShowQuery = {
  title: 'Игра престолов',
  isTvShow: true,
};
const countryId = 122; // Kazakhstan
const cityId = 706; // Almaty
const cinemaId = 281144; // CINEMAX Dostyk Multiplex

describe('Kinopoisk', () => {
  let kp: Kinopoisk;

  beforeAll(() => {
    kp = new Kinopoisk();
  });

  it('fetches movie info for a given id', async () => {
    expect(modelFromObject(await kp.getFilmInfo(filmId))).toMatchSnapshot();
  });

  it('formats movie info response according to graphql query', async () => {
    const res = await kp.getFilmInfo(filmId, `
      {
        kpId
        title
        synopsis
      }
    `);

    expect(modelFromObject(res)).toMatchSnapshot();
  });

  it('finds best movie id for a given query', async () => {
    expect(await kp.getFilmId(movieQuery)).toMatchSnapshot();
  }, 7000);

  it('finds best tv show id for a given query', async () => {
    expect(await kp.getFilmId(tvShowQuery)).toMatchSnapshot();
  }, 7000);

  it('fetches movie credits for a given id', async () => {
    const res = await kp.getFilmCredits(filmId);

    expect(modelFromObject(res)).toMatchSnapshot();
    expect(modelFromFirstElementAtPath('cast')(res)).toMatchSnapshot();
  });

  it('formats movie credits response according to graphql query', async () => {
    const res = await kp.getFilmCredits(filmId, `
      {
        cast { ...MemberProfile }
        crew {
          cinematographers { ...MemberProfile }
        }
      }

      fragment MemberProfile on Member {
        name
        photoUrl
      }
    `);

    expect(modelFromFirstElementAtPath('cast')(res)).toMatchSnapshot();
    expect(
      modelFromFirstElementAtPath('crew.cinematographers')(res),
    ).toMatchSnapshot();
  });

  it('fetches movie gallery for a given id', async () => {
    expect(modelFromObject(await kp.getFilmGallery(filmId))).toMatchSnapshot();
  });

  it('formats movie gallery response according to graphql query', async () => {
    const res = await kp.getFilmGallery(filmId, `
      {
        stills
        posters
      }
    `);

    expect(modelFromObject(res)).toMatchSnapshot();
  });

  it('fetches similar movies for a given id', async () => {
    const res = await kp.getSimilarFilms(filmId);

    expect(modelFromFirstElementAtPath('items')(res)).toMatchSnapshot();
  });

  it('formats similar movies response according to graphql query', async () => {
    const res = await kp.getSimilarFilms(filmId, `
      {
        items {
          kpId
          title
        }
      }
    `);

    expect(modelFromFirstElementAtPath('items')(res)).toMatchSnapshot();
  });

  it('fetches supported countries', async () => {
    const res = await kp.getSupportedCountries();

    expect(modelFromFirstElement(res)).toMatchSnapshot();
  });

  it('fetches supported cities for a given country id', async () => {
    const res = await kp.getSupportedCities(countryId);

    expect(modelFromFirstElement(res)).toMatchSnapshot();
  });

  it('fetches cinemas for a given city id', async () => {
    const res = await kp.getCinemasInCity(cityId);

    expect(modelFromFirstElement(res)).toMatchSnapshot();
  });

  it('fetches cinema info for given arguments', async () => {
    const res = await kp.getCinemaInfo({
      cinemaId,
      date: '14.11.2016',
      utcOffset: '+0600',
    });

    expect(modelFromObject(res)).toMatchSnapshot();
    expect(modelFromFirstElementAtPath('showtimes')(res)).toMatchSnapshot();
    expect(
      modelFromFirstElementAtPath('showtimes.0.items')(res),
    ).toMatchSnapshot();
  });

  it('formats cinema info response according to graphql query', async () => {
    const res = await kp.getCinemaInfo(
      {
        cinemaId,
        date: '14.11.2016',
        utcOffset: '+0600',
      }, `
        {
          id
          name
          showtimes {
            items {
              timestamp
            }
          }
        }
      `,
    );

    expect(modelFromObject(res)).toMatchSnapshot();
    expect(
      modelFromFirstElementAtPath('showtimes.0.items')(res),
    ).toMatchSnapshot();
  });
});
