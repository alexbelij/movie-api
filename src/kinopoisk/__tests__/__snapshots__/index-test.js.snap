exports[`Kinopoisk fetches movie credits from kinopoisk for a given id 1`] = `
Object {
  "cast": Array [
    Object {
      "name": "Джон Бойега",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_2196854.jpg?d=20130703131657",
    },
    Object {
      "name": "Дэйзи Ридли",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_3016071.jpg?d=20140430113652",
    },
    Object {
      "name": "Оскар Айзек",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_43284.jpg",
    },
    Object {
      "name": "Адам Драйвер",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_1833677.jpg?d=20130521201357",
    },
    Object {
      "name": "Донал Глисон",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_671252.jpg?d=20150622214531",
    },
    Object {
      "name": "Харрисон Форд",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_5679.jpg?d=20130506141058",
    },
    Object {
      "name": "Марк Хэмилл",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_23716.jpg?d=20130705122542",
    },
    Object {
      "name": "Кэрри Фишер",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_8459.jpg?d=20130705123958",
    },
    Object {
      "name": "Макс фон Сюдов",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_3311.jpg",
    },
    Object {
      "name": "Энди Серкис",
      "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_13176.jpg",
    },
  ],
  "crew": Object {
    "cinematographers": Array [
      Object {
        "name": "Дэниэл Миндел",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_610174.jpg?d=20150813191602",
      },
    ],
    "composers": Array [
      Object {
        "name": "Джон Уильямс",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_225027.jpg?d=20130908235640",
      },
    ],
    "directors": Array [
      Object {
        "name": "Джей Джей Абрамс",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_27741.jpg?d=20150827134539",
      },
    ],
    "producers": Array [
      Object {
        "name": "Джей Джей Абрамс",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_27741.jpg?d=20150827134539",
      },
      Object {
        "name": "Брайан Берк",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_663746.jpg",
      },
      Object {
        "name": "Кэтлин Кеннеди",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_20396.jpg?d=20150721001100",
      },
      Object {
        "name": "Пиппа Андерсон",
        "photoUrl": null,
      },
      Object {
        "name": "Майкл Арндт",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_936886.jpg?d=20130612003448",
      },
      Object {
        "name": "Лейфур Б. Дагфиннссон",
        "photoUrl": null,
      },
      Object {
        "name": "Томми Гормли",
        "photoUrl": null,
      },
      Object {
        "name": "Томми Харпер",
        "photoUrl": null,
      },
      Object {
        "name": "Джейсон Д. МакГатлин",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_43725.jpg",
      },
      Object {
        "name": "Мишель Рейван",
        "photoUrl": null,
      },
    ],
    "writers": Array [
      Object {
        "name": "Лоуренс Кэздан",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_26537.jpg?d=20131017121035",
      },
      Object {
        "name": "Джей Джей Абрамс",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_27741.jpg?d=20150827134539",
      },
      Object {
        "name": "Майкл Арндт",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_936886.jpg?d=20130612003448",
      },
      Object {
        "name": "Джордж Лукас",
        "photoUrl": "http://st.kp.yandex.net/images/actor_iphone/iphone360_24284.jpg",
      },
    ],
  },
}
`;

exports[`Kinopoisk fetches movie info from kinopoisk for a given id 1`] = `
Object {
  "ageRating": "Number",
  "genres": "Array<String>",
  "imdbRating": "Number",
  "imdbRatingVoteCount": "Number",
  "kpId": "Number",
  "kpRating": "Number",
  "kpRatingVoteCount": "Number",
  "mpaaRating": "String",
  "originalTitle": "String",
  "posterUrl": "String",
  "productionCountries": "Array<String>",
  "rtCriticsRating": "Number",
  "rtCriticsRatingVoteCount": "Number",
  "runtime": "Number",
  "stills": "Array<String>",
  "synopsis": "String",
  "title": "String",
  "year": "Number",
}
`;

exports[`Kinopoisk finds best movie id for a given query 1`] = `714888`;

exports[`Kinopoisk finds best tv show id for a given query 1`] = `464963`;

exports[`Kinopoisk formats movie credits response according to graphql query 1`] = `
Object {
  "cast": "Array<Object>",
  "crew": Object {
    "cinematographers": "Array<Object>",
  },
}
`;

exports[`Kinopoisk formats movie info response according to graphql query 1`] = `
Object {
  "kpId": "Number",
  "synopsis": "String",
  "title": "String",
}
`;
